src/lib/define-dynamic-block.js

```diff
@@ -176,14 +219,64 @@ const defineDynamicBlock = (ScratchBlocks, categoryInfo, staticBlockInfo, extend
                 }
                 return `%${++argCount}`;
             });
+            const wasRendered = this.rendered;
+            this.rendered = false;
+            const connectionMap = this.disconnectOldBlocks_();
+            this.removeAllInputs_();
+
             this.interpolate_(scratchBlocksStyleText, args);
+
+            // Interpolate takes care of laying out the base block,
+            // now we need to reconnect blocks into the inputs
+            // with the same name.
+            let inputIndex = -1;
+            args.forEach(arg => {
+                let type;
+                if (arg.type === 'input_value') {
+                    inputIndex++;
+                    if (arg.check && arg.check === 'boolean') {
+                        type = 'b';
+                    } else {
+                        type = 's';
+                    }
+                    if (inputIndex < this.inputList.length) {
+                        const input = this.inputList[inputIndex];
+                        this.populateArgument_(type, inputIndex, connectionMap, input.name, input);
+                    }
+                }
+            });
```

Comment:
This seems tied to `scratch-blocks` internals even tighter than the rest of the code in this file. If it's necessary to do it this way then c'est la vie but if there's a way to use a scratch-blocks utility for it that would make me sleep better. I imagine you looked, though...

Reply:
The scratch-blocks utilities in question are from the custom procedure code. All of these functions are custom to the custom procedure code, but the good news is that we can make them general utilities in scratch-blocks post-extensionification maybe?

Reply:
This is sort of why we temporarily put extensionifying custom procedures on the back burner...

Reply:
Ah, that makes sense.